src/demo/DemoTestLoanService.java

```diff
@@ -2519,7 +2519,9 @@ private static SavingsAccount makeSavingsAccountForLoanWithSettlements(LoanAccou
 		savingsAccount.setProductTypeKey(savingsProduct.getEncodedKey());
 		savingsAccount.setAccountType(savingsProduct.getProductType());
 		savingsAccount.setAccountState(AccountState.PENDING_APPROVAL);
-		savingsAccount.setInterestRate(new BigDecimal(1.50));
+		if(!savingsProduct.isAllowOffset()){
+			savingsAccount.setInterestRate(new BigDecimal(1.50));
```

Comment:
Do we need to skip setting interest rate for Offset savings here?  Offset product  savings seem to support interest rate settings in the usual way.
OBS: Creating a valid savings account needs lots of checks. This is, of course, is optional but  may be in a future a more generic existent method in SavingsDemo can be enhanced to be used somehow to provide helpers for creating a saving account meeting all product restrictions.